utils/kubernetes/service.go

```diff
@@ -33,12 +33,14 @@ func GetEndpoint(ctx context.Context, opts *ServiceOptions, obj *corev1.Service)
 	var nodePort, clusterPort int32
 	endpoint := utils.Endpoint{}
 
-	if opts.PortSelector != "" {
-		for _, port := range obj.Spec.Ports {
-			if port.Name == opts.PortSelector {
-				nodePort = port.NodePort
-				clusterPort = port.Port
-			}
+	for _, port := range obj.Spec.Ports {
+		if opts.PortSelector != "" && port.Name == opts.PortSelector {
+			nodePort = port.NodePort
```

Comment:
not quite sure I follow the logic here, but if the desired port is the second (or third...) in the list, and `opts.PortSelector != ""` does it actually reach it?

Reply:
The PortSelector would be the name of the port that we would want to connect to, so it would point to the exact item in the list where the `Name` matches the selector. The desired port cannot be based on index, because the order can be varying. So if no PortSelector is specified, it would return the last `Port` element in the list (Which works out well if there is only one port to connect to) @mgfeller

Reply:
Ill work on that, yes sounds good. @mgfeller